tools/Google.Cloud.Tools.GenerateCanonicalLinks.Tests/CanonicalizerTest.cs

```diff
@@ -0,0 +1,50 @@
+        [InlineData("Google.Cloud.CloudBuild.V1", "history.html", "Google.Cloud.CloudBuild.V1/latest/history")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Google.LongRunning.Operation-2.html", "Google.LongRunning/latest/Google.LongRunning.Operation-2")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Google.LongRunning.html", "Google.LongRunning/latest/Google.LongRunning")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Grpc.Core.html", "Grpc.Core/latest/Grpc.Core")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Google.Api.Gax.CloudRunPlatformDetails.html", "Google.Api.Gax/latest/Google.Api.Gax.CloudRunPlatformDetails")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Google.Api.Gax.Grpc.ApiCall-2.html", "Google.Api.Gax/latest/Google.Api.Gax.Grpc.ApiCall-2")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Google.Type.CalendarPeriod.html", "Google.Api.CommonProtos/latest/Google.Type.CalendarPeriod")]
+        [InlineData("Google.Cloud.CloudBuild.V1", "api/Google.Api.Advice.html", "Google.Api.CommonProtos/latest/Google.Api.Advice")]
+        [InlineData("Google.Cloud.Storage.V1", "api/Google.Cloud.Storage.V1.CreateNotificationOptions.html", "Google.Cloud.Storage.V1/latest/Google.Cloud.Storage.V1.CreateNotificationOptions")]
+        [InlineData("Google.Apis.Auth", "api/Google.Apis.Auth.OAuth2.GoogleCredential.html", "Google.Apis/latest/Google.Apis.Auth.OAuth2.GoogleCredential")]
+        [InlineData("Google.Cloud.Spanner.V1", "api/Google.Protobuf.ByteString.html", "Google.Protobuf/latest/Google.Protobuf.ByteString")]
+        [InlineData("Google.Apis.Core", "api/Google.Apis.ISerializer.html", "Google.Apis/latest/Google.Apis.ISerializer")]
+        [InlineData("Google.Apis", "api/Google.Apis.ETagAction.html", "Google.Apis/latest/Google.Apis.ETagAction")]
+        [InlineData("Google.Apis", "api/Google.Apis.Download.DownloadStatus.html", "Google.Apis/latest/Google.Apis.Download.DownloadStatus")]
+        [InlineData("Google.Cloud.Asset.V1", "api/Google.Cloud.Iam.V1.AuditConfigDelta.html", "Google.Cloud.Iam.V1/latest/Google.Cloud.Iam.V1.AuditConfigDelta")]
+        [InlineData("Google.Cloud.Logging.V2", "api/Google.Cloud.Logging.Type.HttpRequest.html", "Google.Cloud.Logging.Type/latest/Google.Cloud.Logging.Type.HttpRequest")]
+        [InlineData("Google.Cloud.Firestore", "api/Google.Cloud.Firestore.V1.Firestore.html", "Google.Cloud.Firestore.V1/latest/Google.Cloud.Firestore.V1.Firestore")]
+        [InlineData("Google.Cloud.Asset.V1", "api/Google.Cloud.OrgPolicy.V1.Policy.html", "Google.Cloud.OrgPolicy.V1/latest/Google.Cloud.OrgPolicy.V1.Policy")]
+        [InlineData("Google.Cloud.Asset.V1", "api/Google.Identity.AccessContextManager.Type.DeviceEncryptionStatus.html", "Google.Identity.AccessContextManager.Type/latest/Google.Identity.AccessContextManager.Type.DeviceEncryptionStatus")]
```

Comment:
Would it make sense to try with something Diagnostics? I'm thinking of two cases (which might already work):
- Something from Diagnostics.Common being used in Diagnostics.AspNetCore.
- And how we have one solution only for Diagnostics.AspNetCore[3] with pragma if for setting one namespace or the other. So maybe try the same type in both Diagnostics.AspNetCore and Diagnostics.AspNetCore3?

Reply:
Ooh, I'd definitely missed Diagnostics.Common, yes. Okay, will add tests for those and then fix them. Thanks!

Reply:
Added Common tests and fixed them... AspNetCore and AspNetCore3 should be okay, but the published docs are currently broken - I need to investigate that separately.